addons/website_forum/tests/test_forum_post.py

```diff
@@ -0,0 +1,45 @@
+from odoo import fields
+from odoo.addons.website_forum.tests.common import TestForumCommon
+
+
+class TestForumPost(TestForumCommon):
+
+    def test_get_related_posts(self):
+        """Test the method returns 5 related posts based on tag similarity."""
+        # Create forum posts and associate them tags
+        forum_tags = self.env['forum.tag'].with_user(self.user_admin).create([
+            {
+                'name': f'tag_{i}',
+                'forum_id': self.forum.id,
+            }
+            for i in range(10)
+        ])
+        forum_posts = self.env['forum.post'].with_user(self.user_admin).create(
+            [
+                {
+                    'content': 'A post ...',
+                    'forum_id': self.forum.id,
+                    'name': 'Post...',
+                    'tag_ids': forum_tags[:i],
+                    'last_activity_date': fields.Datetime.subtract(fields.Datetime.now(), seconds=i)
```

Comment:
Would subtract hours to be closer to real use cases, even if it should not matter.

Although, thinking about DST stuff, maybe it could lead to undeterministic issues.

How about subtracting days actually, just to be safe?